tee-worker/cli/src/trusted_base_cli/commands/litentry/request_vc.rs

```diff
@@ -0,0 +1,349 @@
+}
+
+#[derive(Args)]
+pub struct A13Arg {
+	pub account: String,
+}
+
+#[derive(Subcommand)]
+pub enum OneblockCommand {
+	Completion,
+	Outstanding,
+	Participation,
+}
+
+#[derive(Subcommand)]
+pub enum AchainableCommand {
+	AmountHolding(AmountHoldingArg),
+	AmountToken(AmountTokenArg),
+	Amount(AmountArg),
+	Amounts(AmountsArg),
+	Basic(BasicArg),
+	BetweenPercents(BetweenPercentsArg),
+	ClassOfYear(ClassOfYearArg),
+	DateInterval(DateIntervalArg),
+	DatePercent(DatePercentArg),
+	Date(DateArg),
+	Token(TokenArg),
+}
+
+// I haven't found a good way to use common args for subcommands
+#[derive(Args)]
+pub struct AmountHoldingArg {
+	pub name: String,
+	pub chain: String,
+	pub amount: String,
+	pub date: String,
+	pub token: Option<String>,
+}
+
+#[derive(Args)]
+pub struct AmountTokenArg {
+	pub name: String,
+	pub chain: String,
+	pub amount: String,
+	pub token: Option<String>,
+}
+
+#[derive(Args)]
+pub struct AmountArg {
+	pub name: String,
+	pub chain: String,
+	pub amount: String,
+}
+
+#[derive(Args)]
+pub struct AmountsArg {
+	pub name: String,
+	pub chain: String,
+	pub amount1: String,
+	pub amount2: String,
+}
+
+#[derive(Args)]
+pub struct BasicArg {
+	pub name: String,
+	pub chain: String,
+}
+
+#[derive(Args)]
+pub struct BetweenPercentsArg {
+	pub name: String,
+	pub chain: String,
+	pub greater_than_or_equal_to: String,
+	pub less_than_or_equal_to: String,
+}
+
+#[derive(Args)]
+pub struct ClassOfYearArg {
+	pub name: String,
+	pub chain: String,
+}
+
+#[derive(Args)]
+pub struct DateIntervalArg {
+	pub name: String,
+	pub chain: String,
+	pub start_date: String,
+	pub end_date: String,
+}
+
+#[derive(Args)]
+pub struct DatePercentArg {
+	pub name: String,
+	pub chain: String,
+	pub token: String,
+	pub date: String,
+	pub percent: String,
+}
+
+#[derive(Args)]
+pub struct DateArg {
+	pub name: String,
+	pub chain: String,
+	pub date: String,
+}
+
+#[derive(Args)]
+pub struct TokenArg {
+	pub name: String,
+	pub chain: String,
+	pub token: String,
+}
+
+impl RequestVcCommand {
+	pub(crate) fn run(&self, cli: &Cli, trusted_cli: &TrustedCli) -> CliResult {
+		let alice = get_pair_from_str(trusted_cli, "//Alice");
+		let id: Identity = Identity::from_did(self.did.as_str()).unwrap();
+
+		let (mrenclave, shard) = get_identifiers(trusted_cli);
+		let nonce = get_layer_two_nonce!(alice, cli, trusted_cli);
+
+		let assertion = match &self.command {
+			Command::A1 => Assertion::A1,
+			Command::A2(arg) => Assertion::A2(to_para_str(&arg.guild_id)),
+			Command::A3(arg) => Assertion::A3(
+				to_para_str(&arg.guild_id),
+				to_para_str(&arg.channel_id),
+				to_para_str(&arg.role_id),
+			),
+			Command::A4(arg) => Assertion::A4(to_para_str(&arg.minimum_amount)),
+			Command::A6 => Assertion::A6,
+			Command::A7(arg) => Assertion::A7(to_para_str(&arg.minimum_amount)),
+			Command::A8(arg) => {
+				let networks: Vec<Web3Network> =
+					arg.networks.iter().map(|n| n.as_str().try_into().unwrap()).collect();
```

Comment:
It would be good to improve error handling here. If user provides wrong network it panics like this:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', cli/src/trusted_base_cli/commands/litentry/request_vc.rs:241:71
```

Consider other places too.

Reply:
Thanks - I changed it to `expect(...)` to give more error info